### PR TITLE
clone3: handle clone3() with CLONE_PARENT

### PR DESCRIPTION
**Pull request by adrianreber**
clone3() explicitly blocks setting an exit_signal if CLONE_PARENT is specified. With clone() it also did not work, but there was no error message. The exit signal from the thread group leader is taken.

runc restores are currently failing if clone3() is present as clone3() returns EINVALID if exit_signal is set and CLONE_PARENT is specified.

This fix works for me, but I am not sure if it is the right approach.

### CODE REVIEW

**Comment by adrianreber** [comment]
Also see #926

File: criu/clone-noasan.c [hunk 1]
```diff
@@ -61,7 +61,14 @@ int clone3_with_pid_noasan(int (*fn)(void *), void *arg, int flags,
...
 
-	c_args.exit_signal = exit_signal;
+	/*
+	 * clone3() explicitly blocks setting an exit_signal
+	 * if CLONE_PARENT is specified. With clone() it also
+	 * did not work, but there was no error message. The
+	 * exit signal from the thread group leader is taken.
+	 */
+	if (!(flags & CLONE_PARENT))
+		c_args.exit_signal = exit_signal;
```
Comment (avagin) [inline criu/clone-noasan.c]: should we failt if exit_signal isn't equal to SIGCHLD?

Comment (adrianreber) [reply criu/clone-noasan.c]: No idea, but if you think it would be correct, I can add that. Let me add it and push a new version of this PR.

Comment (avagin) [reply criu/clone-noasan.c]: Yes, I think it would be correct.

Comment (adrianreber) [reply criu/clone-noasan.c]: Now failing if `exit_signal != SIGCHLD`

**Pull request merged**